api/authentication.py

```diff
@@ -42,6 +48,12 @@ def introspect_token(introspect_token_url, token):
 
 
 class FxaTokenAuthentication(BaseAuthentication):
+    def authenticate_header(self, request):
+        # Note: we need to implement this function to make DRF return a 401 status code
+        # when we raise AuthenticationFailed, rather than a 403.
+        # See https://www.django-rest-framework.org/api-guide/exceptions/#authenticationfailed
```

Comment:
https://www.django-rest-framework.org/api-guide/authentication/#custom-authentication seemed more relevant to me